### PR TITLE
clarified ux; moved lists to cards; added many tutorials

### PR DESCRIPTION
**Pull request by aaronmarkham**
* Renamed `module` folder to `modules` as the contents are plural and it confused me at first that it was about the old Module API.
* Main Python guide page: http://204.236.210.14/guide/index.html
    - Added "core APIs" like ndarray and autograd 
    - Promoting Gluon as first "getting started" card
* Gluon guide page: http://204.236.210.14/guide/modules/gluon/index.html
    - The [list view of the tutorials was also confusing](https://beta.mxnet.io/guide/module/gluon/index.html), so I switched them to [cards](http://204.236.210.14/guide/modules/gluon/index.html)
    - I updated the Gluon section to follow the flow from the [wireframes](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=103089084). 
    - We need an inference section, but without tutorials to link to here, I'm leaving that out for now.
* NDArray tutorials updated from a [list view of one tutorial](https://beta.mxnet.io/guide/module/ndarray/index.html), to [cards showing two tutorials](http://204.236.210.14/guide/modules/ndarray/index.html).
    - This is important since we have so many other ndarray tutorials to port over, they need to be separated and not in some cluttered list view.
* Modules index is orphaned for now and still has the kind of confusing list view: http://204.236.210.14/guide/modules/index.html
    - When there are more APIs/modules covered and we want to clear some space on the main Python Guide page, I recommend we move those cards to the module index and link to it from the Python Guide page. (like in the wireframes). For now there aren't that many so bringing them forward is good.
* I updated some of the language for clarity and grammar.
* Fixed some toctree entries where pages were not listed and causing warnings.
* Added Performance tutorials
* Added Deployment tutorials
* Added Customization tutorials
* Added Image and Text tutorials - these are on a separate line as I think when we do a redesign these should be grouped separately.
* I changed some files from using dashes to underscore for consistency.

I still expect some reorganization when the API section gets more attention, but this clears things up a bit at least for the guide and tutorials, paving the way for porting over more of them from the existing website.

### CODE REVIEW

File: python/guide/index.rst [hunk 1]
```diff
@@ -7,27 +7,61 @@ Getting started
...
 
-      A quick overview of the core concepts of MXNet.
+      MXNet's imperative interface for Python.
 
    .. card::
       :title: For PyTorch users
       :link: to-mxnet/pytorch.html
 
-      Compare MXNet and PyTorch on implementing a simple handwritten digits classifier.
+      Compares MXNet to PyTorch on implementing MNIST.
```
Comment (astonzhang) [inline python/guide/index.rst]: Compares MXNet with PyTorch?

**Comment by astonzhang** [comment]
On how we'd like to organize tutorials on guide/:

1. Items below are what we would like to have for the first release of the new website at http://beta.mxnet.io/guide/index.html
2. Items are usually the names of the existing files. For instance, 1-ndarray refers to https://github.com/mli/new-docs/blob/master/python/guide/crash-course/1-ndarray.md. If such a file does not exist, such as Symbol, create it.
3. For items with explanations of “need more details”
    1. Add more text descriptions
    2. Explain API usage with more examples
    3. Add contents from D2L (if any)
4. For items without explanations, polish them.
5. For optional items, deliver them if we have time.


* Getting Started
    * A 60-minute crash course
        * 1-ndarray
        * 2-nn
        * 3-autograd
        * 4-train
        * 5-predict
        * 6-use_gpus
    * Moving from other DL frameworks
        * to-mxnet/pytorch.html
        * to-mxnet/tf.html
* Packages and Modules
    * ndarray
        * ndarray (need more details, refer to https://en.d2l.ai/chapter_crashcourse/ndarray.html)
        * sparse
    * autograd (need more details)
    * Using GPU (refer to https://en.d2l.ai/chapter_deep-learning-computation/use-gpu.html)
    * optimizer
    * lr_scheduler
    * Symbol
    * gluon
        * nn (need more details, refer to https://en.d2l.ai/chapter_deep-learning-computation/model-construction.html)
        * parameters (need more details, refer to https://en.d2l.ai/chapter_deep-learning-computation/parameters.html)
        * init (need more details, refer to https://en.d2l.ai/chapter_deep-learning-computation/parameters.html and https://en.d2l.ai/chapter_deep-learning-computation/deferred-init.html)
        * custom-layer (need more details, refer to https://en.d2l.ai/chapter_deep-learning-computation/custom-layer.html)
        * data
        * image_augmentation (should show how to use gluon.data.vision.transforms.Compose, refer to https://en.d2l.ai/chapter_computer-vision/image-augmentation.html)
        * file I/O (refer to https://en.d2l.ai/chapter_deep-learning-computation/read-write.html)
        * loss (should explain how to create customized loss)
        * trainer
        * hybridize (need more details, refer to https://en.d2l.ai/chapter_computational-performance/hybridize.html)
        * model_zoo
* Performance
    * Essentials
        * (optional) perf
        * profiler
    * Compression
        * float16 
        * (optional) int8
    * Accelerated Backend
        * (optional) tensor-rt
* Deployment
    * Deploy into Another Language
        * cpp 
        * scala/java
        * ONNX
* Customization (extend)
    * customop (should address customer complaints: https://discuss.gluon.ai/t/topic/10822)

**Comment by aaronmarkham** [comment]
@astonzhang Thanks for the list. After re-reading that, I can see that I've accomplished several of the bullets in this PR. The ones not finished should not be a reason to hold back a merge of this PR. Besides, a lot of the unfinished stuff just doesn't exist yet, so there's no way to "link it up" or "port it over".

I'll flag some of the items in the list as complete, so it's easier to compare.

**Comment by astonzhang** [comment]
Yes, we don't need to cover all the items of the list in this PR. However, for those existing items, could you rename/reorganize based on the list?

For example, for to-mxnet/pytorch.html, could you give a title "Moving from other DL frameworks" and include to-mxnet/pytorch.html in to-mxnet/index? For gluon, we may remove "Getting started" and reorder the items based on the list (earlier items are necessary to understand later items).

**Comment by aaronmarkham** [comment]
I made your suggested changes plus converted all of the md files that were lingering in there to rst and put them in the guide.
Before this PR gets too unwieldy, let's merge please. LMK if you have any other suggestions.

**Comment by astonzhang** [comment]
Thanks Aaron. I noticed that there are still some inconsistency, such as whether to put customer-layer under gluon/ (in the list) or extend/ (in this PR), and whether to include run-on-aws (in this PR, but not in the list).

I would prefer that we follow the list. For instance, some contents of run-on-aws seem obsolete. If further working on this PR blocks you, you may merge this PR then revise in a separate PR later.

**Comment by aaronmarkham** [comment]
I've been following the list but using what was already loaded in the repo as .md files. I didn't include the cloud.rst file in the navigation, because, yes, it seems old, and the other files seem to cover the topic better. Also, cloud should be any cloud provider and we do have this info on the production site now. I can port that over later...

I'm going to fix some navigation issues I spotted in the preview, then merge and fix more things in another PR.

**Comment by thomelane** [comment]
A great step in the right direction @aaronmarkham! Some thoughts.

* Still not happy about 'Modules'. Can we just stick with 'Packages'? We've technically got 'Modules and Packages' being discussed in this section, but do we need to be that technically correct? We both got confused with Module of MXNet vs Module API, and we work with MXNet daily!

* We have a weird user flow in some cases with higher level packages (e.g. learning rate and autograd). We have a nice flow inside of Gluon section (i.e. data -> nn -> init -> ...) but it's missing key parts like autograd just because that's a higher level shared api. Code structure has lead to this I realise. Multiple links to the same place could be a quick fix.

* Sections under Guide lead to some inconsistency: e.g. training is in Gluon section, but deployment is straight under Guide. Worth adding the training section to Guide?

And some minor things that can be resolved in the future.

* Can’t 'open in new tab' on cards.
* We don't yet have breadcrumbs.
* Should have an intro to Guide.

**Comment by aaronmarkham** [comment]
@thomelane 

> * Still not happy about 'Modules'. Can we just stick with 'Packages'? We've technically got 'Modules and Packages' being discussed in this section, but do we need to be that technically correct? We both got confused with Module of MXNet vs Module API, and we work with MXNet daily!

Agree - if @astonzhang also agrees, let's make this change as it will help with clarity. I'll rename the folder too.

> * We have a weird user flow in some cases with higher level packages (e.g. learning rate and autograd). We have a nice flow inside of Gluon section (i.e. data -> nn -> init -> ...) but it's missing key parts like autograd just because that's a higher level shared api. Code structure has lead to this I realise. Multiple links to the same place could be a quick fix.

Yes, I repeat the cards in different places when they make sense contextually - like Learning Rate is in the modules area and in the gluon area. Let me know specifically what to add and where in an issue and I'll make it happen.


> * Sections under Guide lead to some inconsistency: e.g. training is in Gluon section, but deployment is straight under Guide. Worth adding the training section to Guide?

Yes, we should have training there... and then maybe different cards for training using different APIs.
Right now we only have Gluon, but I can add that so it is part of the navigation and it'll make a lot of sense. I think we should also have inference as a main navigation item in the guide too. WDYT?

> 
> And some minor things that can be resolved in the future.
> 
> * Can’t 'open in new tab' on cards.
> * We don't yet have breadcrumbs.
> * Should have an intro to Guide.

I added issues for breadcrumbs and "open in tab". 
I added the intro task to the roadmap.

**Comment by astonzhang** [comment]
Although technically we should call it "Modules and Packages", I am OK to not to mention "Modules" to avoid confusion with MXNet Module API
